REFERENCE.md

```diff
@@ -0,0 +1,363 @@
+
+Remote remote = new Remote(requestConfig);
+// Call get() or post() with a URL:
+remote.get("http://schemas.learnosity.com/stable/questions/templates");
+
+// getBody() gives you to body of the request
+String body = response.getBody();
+```
+
+#### Arguments
+
+**URL**<br>
+A string URL, including schema and path. Eg:
+
+```
+https://schemas.learnosity.com/stable/questions/templates
+```
+
+**Data**<br>
+An optional array of data to be sent as a payload. For GET it will be a URL encoded query string.
+
+### Remote methods
+The following methods are available after making a `get()` or `post()`.
+
+**getBody()**<br>
+Returns the body of the response payload.
+
+**getHeader()**<br>
+Currently only returns the *content_type* header of the response.
+
+**getStatusCode()**<br>
+Returns the HTTP status code of the response.
+
+<hr>
+
+### DataApi
+
+This is a helper class for use with the Data API. It creates the initialisation packet and sends a request to the Data API, returning a JSONObject with the response data. There is also a requestRecursive function which can be called with a class implementing RequestCallback.java. The execute() function will be called for each response.
+
+The DataApi Constructor can handle either 3 arguments;
+* url
+* securityPacket
+* secret
+
+or 4 arguments:
+* url
+* securityPacket
+* secret
+* action
+
+or 5 arguments:
+* url
+* securityPacket
+* secret
+* requestPacket
+* action
+
+```java
+Map<String,String> sec = new HashMap<String, String>();
+sec.put("consumer_key", "yis0TYCu7U9V4o7M");
+sec.put("domain","localhost");
+
+reqData = new HashMap<String,String>();
+reqData.put("limit", "10");
+
+DataApi dataApi = new DataApi("https://data.learnosity.com/latest/itembank/items", sec, consumerSecret, reqData, "get");
+
+// you can also configure the Data API with a requestConfig as above
+RequestConfig requestConfig = RequestConfig.custom().setSocketTimeout(1000).build();
+
+dataApi.setRequestConfig(requestConfig);
+
+
+JSONObject response = dataApi.requestJSONObject();
+JSONObject res = new JSONObject(response.getString("body"));
+
+// You can also get a Remote object like this:
+Remote remote = dataApi.request();
+String body = remote.getBody();
+```
+
+## Troubleshooting
+
+We recommend that you use the Java JSON implementation which comes packaged in the sdk. In the past, we had several issues which were due to incompatible JSON implementations.
+
+
+## Creating jar files (using Eclipse Luna Release 4.4.0)
+
+The 'light' jar:
+
+In eclipse right click the java sdk project and select 'Export'. Under 'Java', select 'JAR file'. In the following screen only select the packages in the src (excluding the library folder) and click 'Finish'.
+
+The 'all' jar:
+
+In eclipse right click the java sdk project and select 'Export'. Under 'Java', select 'Runnable JAR file', click 'Next' and then 'Finish'. There are warnings that some libararies are included. This can be disregarded as the libraries are publicly available.
```

Comment:
```suggestion
In Eclipse, right-click the Java SDK project and select 'Export'. Under 'Java', select 'Runnable JAR file', click 'Next' and then 'Finish'. There are warnings that some libraries are included. This can be disregarded as the libraries are publicly available.
```